rosidl_generator_py/msg/Strings.msg

```diff
@@ -1,5 +1,9 @@
 string empty_string
 string def_string "Hello world!"
+string def_string_delimiter 'Hello"world!'
+string def_string_delimiter2 "Hello'world!"
+string def_string_delimiter3 'Hello'world!'
+string def_string_delimiter4 "Hello"world!"
```

Comment:
ok it does make more sense now. I can remove these two tests and will ticket the expected behavior of the `rosidl_parser` elsewhere. I'll address the change of behavior of the parser along with implementing parsing for arrays if there is no objection.

I guess that the following (without outer quotes) should be invalid too ?
 ```
string def_string_delimiter3 Hello"world!
string def_string_delimiter4 Hello'world!
```
but 
```
string def_string_delimiter3 Hello\"world!
string def_string_delimiter4 Hello\'world!
```
should be valid.

Is that correct?